### PR TITLE
Hierarchical account: account module implementation #456

### PR DESCRIPTION
**Pull request by matias-aguero-hs**

### CODE REVIEW

File: account/src/main/java/org/killbill/billing/account/api/DefaultAccount.java [hunk 1]
```diff
@@ -28,6 +28,8 @@
 import org.killbill.billing.catalog.api.Currency;
 import org.killbill.billing.entity.EntityBase;
 
+import com.sun.org.apache.xpath.internal.operations.Bool;
```
Comment (pierre) [inline account/src/main/java/org/killbill/billing/account/api/DefaultAccount.java]: Wrong import.

File: util/src/test/java/org/killbill/billing/mock/api/MockAccountUserApi.java [hunk 2]
```diff
@@ -167,4 +167,10 @@ public void updateAccount(final UUID accountId, final AccountData accountData, f
             throws AccountApiException {
         throw new UnsupportedOperationException();
     }
+
+    @Override
+    public List<Account> getChildrenAccounts(final UUID uuid, final TenantContext tenantContext) throws AccountApiException {
+        // TODO implement
```
Comment (pierre) [inline util/src/test/java/org/killbill/billing/mock/api/MockAccountUserApi.java]: If you don't need it for now, could you `throw new UnsupportedOperationException();`?

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java [hunk 3]
```diff
@@ -1207,4 +1207,33 @@ public Response removeEmail(@PathParam(ID_PARAM_NAME) final String id,
...
+    // -------------------------------------
+
+    @TimedResource
+    @GET
+    @Path("/{accountId:" + UUID_PATTERN + "}/" + CHILDREN)
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "List children accounts", response = AccountJson.class, responseContainer = "List")
+    @ApiResponses(value = {@ApiResponse(code = 400, message = "Invalid parent account id supplied"),
+                           @ApiResponse(code = 404, message = "Parent Account not found")})
+    public Response getChildrenAccounts(@PathParam("accountId") final String parentAccountId,
```
Comment (pierre) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/AccountResource.java]: Could you add a test to verify the Java client implementation?

**Comment by matias-aguero-hs** [comment]
Suggestions are fixed.

**Pull request merged**